lib/core/src/display/navigation/animation_manager.rs

```diff
@@ -0,0 +1,23 @@
+// ========================
+// === AnimationManager ===
+// ========================
+
+pub struct AnimationManager {
+    maximum_frame_duration: f32
+}
+
+impl AnimationManager {
+    pub fn new(minimum_fps : f32) -> Self {
+        let maximum_frame_duration = 1.0 / minimum_fps;
+        Self { maximum_frame_duration }
+    }
+
+    /// Runs the closure guaranteeing it runs at `minimum_fps`.
+    pub fn run<F:FnMut(f32)>(&mut self, mut dt:f32, mut f:F) {
+        while dt > self.maximum_frame_duration {
+            f(self.maximum_frame_duration);
+            dt -= self.maximum_frame_duration;
+        }
+        f(dt);
```

Comment:
In my previous review I was asking about a situation when a frame takes `34` ms or something like that. I don't see this comment anywhere now, but I would like to talk about this case and understand if this function behaves correctly. I still believe it is not (the same explanation that I provided there).